subnet/kube/cluster_cidr.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 flannel authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"net"
+
+	"github.com/flannel-io/flannel/subnet"
+	"golang.org/x/net/context"
+	networkingv1alpha1 "k8s.io/api/networking/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	log "k8s.io/klog"
+)
+
+func (ksm *kubeSubnetManager) handleAddClusterCidr(obj interface{}) {
+	cluster := obj.(*networkingv1alpha1.ClusterCIDR)
+	if cluster == nil {
+		log.Errorf("received wrong object: %s", obj)
+		return
+	}
+	if cluster.Spec.IPv4 != "" {
+		log.Infof("handleAddClusterCidr: registering CIDR [ %s ]\n", cluster.Spec.IPv4)
+		_, cidr, err := net.ParseCIDR(cluster.Spec.IPv4)
+		if err != nil {
+			log.Errorf("error reading cluster spec: %s", err)
+			return
+		}
+		ksm.subnetConf.AddNetwork(cidr)
+	}
+	if cluster.Spec.IPv6 != "" {
+		log.Infof("handleAddClusterCidr: registering CIDR [ %s ]\n", cluster.Spec.IPv6)
+		_, cidr, err := net.ParseCIDR(cluster.Spec.IPv6)
+		if err != nil {
+			log.Errorf("error reading cluster spec: %s", err)
+			return
+		}
+		ksm.subnetConf.AddNetwork(cidr)
+	}
+
+	err := subnet.WriteSubnetFile(ksm.snFileInfo.path, ksm.subnetConf, ksm.snFileInfo.ipMask, ksm.snFileInfo.sn, ksm.snFileInfo.IPv6sn, ksm.snFileInfo.mtu)
+	if err != nil {
+		log.Errorf("error writing subnet file: %s", err)
+		return
+	}
+}
+
+func (ksm *kubeSubnetManager) handleUpdateClusterCidr(obj, newobj interface{}) {
+	log.Info("starting handleUpdateClusterCidr...")
+	cluster := obj.(*networkingv1alpha1.ClusterCIDR)
+	newcluster := newobj.(*networkingv1alpha1.ClusterCIDR)
+
+	//only look at mutable fields
+	if cluster != nil && newcluster != nil {
+
+		log.Infof("==== handleUpdateClusterCidr[%s]: list Annotations ====", cluster.Name)
+		log.Infof("obj annotations: %v", cluster.Annotations)
+		log.Infof("newobj annotations: %v", cluster.Annotations)
+
+		log.Infof("==== handleUpdateClusterCidr[%s]: list Labels ====", cluster.Name)
+		log.Infof("obj labels: %v", cluster.Labels)
+		log.Infof("newobj labels: %v", cluster.Labels)
+
+		log.Infof("==== handleUpdateClusterCidr[%s]: list Finalizers ====", cluster.Name)
+		log.Infof("obj Finalizers: %v", cluster.Finalizers)
+		log.Infof("newobj Finalizers: %v", cluster.Finalizers)
+	}
+	log.Info("end handleUpdateClusterCidr")
```

Comment:
yes it was just for debugging. there are still updates that are notified because labels and annotations can change but since we don't use them at the moment, I removed the function.